galpy/df/quasiisothermaldf.py

```diff
@@ -1685,6 +1688,194 @@ def sampleV(self,R,z,n=1):
         else:
             return out
 
+    @potential_physical_input
+    def sampleV_interpolate(self,R,z,R_pixel,z_pixel,num_std=3,R_min=None,
+                            R_max=None,z_max=None):
+        """
+        NAME:
+            
+            sampleV_interpolate
+    
+        PURPOSE:
+            
+            Given an array of R and z coordinates of stars, return the 
+            positions and their radial, azimuthal, and vertical velocity.
+    
+        INPUT:
+            
+            R - array of Galactocentric distance (can be Quantity)
+
+            z - array of height (can be Quantity)
+            
+            R_pixel, z_pixel= the pixel size for creating the grid for
+                   interpolation (in natural unit)
+            
+            num_std= number of standard deviation to be considered outliers
+                      sampled separately from interpolation
+                      
+            R_min, R_max, z_max= optional edges of the grid
+    
+        OUTPUT:
+            
+            coord_v= a numpy array containing the sampled velocity, (vR, vT, vz),
+                     where each row correspond to the row of (R,z)
+    
+        HISTORY:
+            
+            2018-08-10 - Written - Samuel Wong (University of Toronto)
+            
+        """
+        #Initialize output array
+        coord_v= numpy.empty((numpy.size(R), 3))
+        #Since the sign of z doesn't matter, work with absolute value of z
+        z= numpy.abs(z)
+        #Separate the coodinates into outliers and normal points.
+        mean_R= numpy.mean(R)
+        std_R= numpy.std(R)
+        mean_z= numpy.mean(z)
+        std_z= numpy.std(z)
+        #Define outliers as points either more than <num_std> std away from
+        #mean, or those outside user-specified grid
+        if R_min is None:
+            R_min_temp= 0
+        else:
+            R_min_temp= R_min
+        if R_max is None:
+            R_max_temp= numpy.max(R)
+        else:
+            R_max_temp= R_max
+        if z_max is None:
+            z_max_temp= numpy.max(z)
+        else:
+            z_max_temp= z_max
+        mask= numpy.any([numpy.abs(R - mean_R) > num_std*std_R, 
+                       numpy.abs(z - mean_z) > num_std*std_z,
```

Comment:
I don't understand why you keep the ``numpy.abs(R - mean_R) > num_std*std_R`` and ``numpy.abs(z - mean_z) > num_std*std_z`` conditions here. Why not do something like (note that I also just include the mean calc. here, so that makes a few of the earlier lines unnecessary)
```
if R_min is None:
    R_min= numpy.mean(R) - num_std*numpy.std(R)
if R_max is None:
    R_max= numpy.mean(R) + num_std*numpy.std(R)
if z_max is None:
    z_max= numpy.mean(z) + num_std*numpy.std(z)
mask= numpy.any([R < R_min, R > R_max, z > z_max],axis = 0)
```
and then just keep ``R_min`` etc. to build the grid. I think you're worried that ``R_min`` might be much less than the actual minimum R of the normal points? But I think we should just ignore that possibility.

Reply:
Yes, I am worried about that. Right now, `num_std` is set to a default value of 3. So most of the time, there are no outliers, and the maximum of normal points is usually smaller than `numpy.mean(R) + num_std*numpy.std(R)`. So I fear this assumption would slow it down.

Reply:
Okay, but then maybe just do
```
R_min= numpy.amax([numpy.mean(R)-num_std*numpy.std(R),numpy.amin(R)])
```
and similar?
and similar?